pkg/bpf/bpf_restart.go

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright 2024 The Kmesh Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at:
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package bpf
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"syscall"
+
+	"github.com/cilium/ebpf"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"kmesh.net/kmesh/pkg/utils"
+	"kmesh.net/kmesh/pkg/version"
+)
+
+const (
+	NewStart = iota
+	Restart
+	Update
+)
+
+const (
+	daemonSetName = "kmesh"
+	namespace     = "kmesh-system"
+)
+
+var kmeshStartStatus int
+
+func GetKmeshStartStatus() int {
+	return kmeshStartStatus
+}
+
+func SetKmeshStartStatus(Status int) {
+	kmeshStartStatus = Status
+}
+
+func GetDaemonset() bool {
```

Comment:
To be more clear, we can move this implement detail into a function `InferRestartStatus`